velox/connectors/hive/storage_adapters/hdfs/HdfsWriteFile.cpp

```diff
@@ -78,7 +84,10 @@ void HdfsWriteFile::append(std::string_view data) {
 
 uint64_t HdfsWriteFile::size() const {
   auto fileInfo = hdfsGetPathInfo(hdfsClient_, filePath_.c_str());
-  return fileInfo->mSize;
+  uint64_t size = fileInfo->mSize;
+  // should call hdfsFreeFileInfo to avoid memory leak
+  hdfsFreeFileInfo(fileInfo, 1);
```

Comment:
This should be in destructor or `close()`

Reply:
There are some difference with HdfsReadFile, when we write some date, we want to get the latest size, so a new hdfsGetPathInfo is need, do you think so?

Reply:
OK I see